api/src/org/apache/cloudstack/api/command/user/resource/ListResourceLimitsCmd.java

```diff
@@ -96,4 +117,24 @@ public void execute() {
         response.setResponseName(getCommandName());
         this.setResponseObject(response);
     }
+
+    private Resource.ResourceType getResourceTypeEnum() {
+        // Map resource type
+        Resource.ResourceType resourceTypeResult = null;
+        if (resourceTypeName != null) {
+            try {
+                resourceTypeResult = Resource.ResourceType.valueOf(resourceTypeName);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidParameterValueException("Please specify a valid resource type name.");
+            }
+        } else if (resourceType != null) {
+            try {
+                resourceTypeResult = Resource.ResourceType.values()[resourceType];
```

Comment:
The code here assumes that the ResourceType.getOrdinal() is always in the same order as the enum values.
This might at one moment not be the case..
I would prefer to search for the ResourceType returning the correct value when calling type.getOrdinal()

Reply:
Correct, I updated the change to use the ordinal value use in the each value.